spec/v3/spec.md

```diff
@@ -1,110 +1,460 @@
+
+### 3.1 Overview
+```
+Offset     00   01   02   03   04   05   06   07   08   09   0A   0B   0C   0D   0E   0F
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000000   |           Magic Number           |  V |         Root Directory Offset         |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000010   |         Root Directory Length         |            Metadata Offset            |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000020   |            Metadata Length            |        Leaf Directories Offset        |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000030   |        Leaf Directories Length        |            Tile Data Offset           |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000040   |            Tile Data Length           |         Num of Addressed Tiles        |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000050   |         Number of Tile Entries        |        Number of Tile Contents        |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000060   |  C | IC | TC | TT |MinZ|MaxZ|              Min Position             |      Max 
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+000070    Position                     |CenZ|            Center Position            |
+         +----+----+----+----+----+----+----+----+----+----+----+----+----+----+----+
+```
+
+### 3.2 Fields
+#### Magic Number
+
+The magic number is a fixed 7-byte field whose value is always `PMTiles` in UTF-8 encoding (`0x50 0x4D 0x54 0x69 0x6C 0x65 0x73`)
+
+#### Version (V)
+
+The version is a fixed 1-byte field whose value is always 3 (`0x03`).
+
+#### Root Directory Offset
+
+The Root Directory Offset is an 8-byte field whose value gives the offset of the first byte of the root directory. This address offset is relative to the first byte of the archive.
+
+This field is encoded as a little-endian 64-bit unsigned integer.
+
+#### Root Directory Length
+
+The Root Directory Length is an 8-byte field specifying the number of bytes in the root directory.
+
+This field is encoded as a little-endian 64-bit unsigned integer.
+
+#### Metadata Offset
+
+The Metadata Offset is an 8-byte field whose value gives the offset of the first byte of the metadata. This address offset is relative to the first byte of the archive.
+
+This field is encoded as a little-endian 64-bit unsigned integer.
+
+#### Metadata Length
+
+The Metadata Length is an 8-byte field specifying the number of bytes reserved for the metadata. 
```

Comment:
Here and in other sections we should never use the word "reserved". Reserved implies that we might have allocated more bytes than is necessary. For example if there is 100 bytes of valid metadata, we never want there be 101 bytes, or 200 bytes. A parser should not be expected to deal with extra zero or irrelevant bytes past the end of the data; and extra bytes at the end don't serve any purpose in the current design as no affordances are provided for overwriting of data in-place.

Reply:
Addressed in c3a6c6ffe